core/annotations.h

```diff
@@ -48,7 +48,7 @@
  * @brief Annotation handler registration routines.
  */
 
-#ifdef CLIENT_INTERFACE
+# ifdef CLIENT_INTERFACE
```

Comment:
I was wondering about exactly the same thing. It's not trivial to detect whole-file ifdefs using the token based parser only. I think we should go with marking whole-file ifdefs with `/* around entire file */ ` as you suggested.